onnxruntime_extensions/tools/pre_post_processing/steps/nlp.py

```diff
@@ -0,0 +1,508 @@
+
+
+class BertTokenizerQATask(Step):
+    def __init__(self, name: Optional[str] = None):
+        """
+        Brief:
+            Just duplicate input_ids for decoder(TokenizerDecoder). For tasks like 'BertTokenizerQATaskDecoder' which need to use the same input_ids for decoder.
+            However, input_ids has its consumers, it will merged and removed in the next step. So we need to duplicate one.
+            The new output 'input_ids_1' will be kept as a new output in graph.
+        Args:
+            name: Optional name of step. Defaults to 'BertTokenizerQATask'
+
+        """
+        super().__init__(
+            ["input_ids", "attention_mask", "token_type_ids"],
+            ["input_ids", "attention_mask", "token_type_ids", "input_ids_1"],
+            name,
+        )
+
+    def _create_graph_for_step(self, graph: onnx.GraphProto, onnx_opset: int):
+        graph_input_names = [inp.name for inp in graph.output]
+        for inp in self.input_names:
+            assert inp in graph_input_names
+
+        # reorder inputs as we assumed input_names[0] is input_ids
+        if "input_ids" not in self.input_names[0]:
+            o_inputs = graph_input_names.copy()
+            for inp in o_inputs:
+                if "input_ids_1" in inp:
+                    self.input_names[0] = inp
+                elif "attention_mask" in inp:
+                    self.input_names[1] = inp
+                else:
+                    self.input_names[2] = inp
+
+        input_type_str0, input_shape_str0 = self._get_input_type_and_shape_strs(graph, 0)
+        output_shape_str = "batch_size,max_seq_len"
+
+        def build_output_declare():
+            output_base = []
+            for out in self.output_names:
+                output_base.append(f"{input_type_str0}[{input_shape_str0}] {out}")
+
+            return ",".join(output_base)
+
+        def build_output_imp():
+            output_str = []
+            for idx, out in enumerate(self.output_names):
+                output_str.append(f"{out} = Identity({self.input_names[idx % len(self.input_names)]})")
+
+            output_str = "\n".join(output_str)
+            return output_str
+
+        def build_input_declare():
+            inputs = []
+
+            for idx, inp in enumerate(self.input_names):
+                input_type_str_x, input_shape_str_x = self._get_input_type_and_shape_strs(graph, idx)
+                inputs.append(f"{input_type_str_x}[{input_shape_str_x}] {inp}")
+
+            return ",".join(inputs)
+
+        def build_tokenizer_call_arg():
+            call_args = f"{self.input_names[0]}"
+            return call_args
+
+        converter_graph = onnx.parser.parse_graph(
+            f"""\
+            qa_task ({build_input_declare()}) 
+                => ({build_output_declare()})
+            {{
+                {build_output_imp()}
+            }}
+            """
+        )
+
+        return converter_graph
+
+
+class BertTokenizerQATaskDecoder(Step):
```

Comment:
nit: the 'Task' in a 'Step' seems a superfluous. Would BertTokenizerQADecoder be fine?